src/api.rs

```diff
@@ -106,6 +106,13 @@ pub trait ElectrumApi {
     /// already subscribed to the script.
     fn script_subscribe(&self, script: &Script) -> Result<Option<ScriptStatus>, Error>;
 
+    /// Batch version of [`script_subscribe`](#method.script_subscribe).
+    ///
+    /// Takes a list of scripts and returns a list of script status responses.
+    fn batch_script_subscribe<'s, I>(&self, scripts: I) -> Result<Vec<Option<ScriptStatus>>, Error>
+    where
+        I: IntoIterator<Item = &'s Script> + Clone;
```

Comment:
The problem in taking `&[&Script]` is that sometimes (eg. reading from a db) you have something iterable that is not necessarily a slice.
But you are right that clone could potentially allocate, @afilini come up with this signature:

```
fn batch_script_subscribe<'s, I, B>(&self, scripts: I) -> Result<Vec<Option<ScriptStatus>>, Error>
    where
        I: IntoIterator<Item = B> + Copy,
        B: Borrow<&'s Script>;
```
What do you think?

Reply:
Also if you have a `Vec<Script>` you need to allocate a new vec of references just to be able to use this method, which is not ideal.

That horrible `IntoIterator<Item = Borrow<&Script>>` should work fine with at least:
- `&Vec<Script>`
- `&Vec<&Script>`

and any other type that is copy

Reply:
> The problem in taking &[&Script] is that sometimes (eg. reading from a db) you have something iterable that is not necessarily a slice.

I see, then the current implementation is probably the most versatile.

I don't think that swapping the iterator bound from `Clone` to `Copy` would work since most iterators, and especially most (if not all) iterator adapters like `Map`, `Filter`, etc. are not copy, so that bound would be overly restrictive.

For example, this wouldn't compile

```rust
let client: Client = todo!();
let scripts: &[&Script] = todo!();
let _ = client.batch_script_subscribe(scripts.iter());
```

Reply:
So maybe we should switch back to `Clone`, but I would keep the `Borrow` thing so that you can use collections of both `Script` and `&Script` directly.

Compared to `Copy` with `Clone` you *can* use the API in a suboptimal way (e.g. by passing in a `Vec<&Script>`, which would allocate a new `Vec`), but as you said `Copy` is too restrictive.

See an example here for how it could be used: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c64e982d7806995dfa2a2dff79265a4d